File: tests/XMP/global-view/structure/C/gmove.c
#include <stdio.h>
#include <stdlib.h>
#include <xmp.h>
#define XMP_FALSE 0
#define XMP_TRUE  1
#pragma xmp nodes p[2]
#pragma xmp template t[10]
#pragma xmp distribute t[block] onto p
struct b{
  int a[10];
};
int a[10];
#pragma xmp align b.a[i] with t[i]
#pragma xmp align a[i] with t[i]
struct b c;

int main(){
#pragma xmp loop on t[i]
  for(int i=0;i<10;i++){
    c.a[i] = i;
    a[i] = i + 100;
  }

#pragma xmp gmove
  a[0:3] = c.a[5:3];

  int flag = XMP_FALSE;
  if(xmpc_node_num() == 0){
    if(a[0] == 5 && a[1] == 6 && a[2] == 7 && a[3] == 103 && a[4] == 104)
      flag = XMP_TRUE;
  }
  
#pragma xmp reduction(+:flag)
    if(flag == XMP_TRUE){
      if(xmpc_node_num() == 0) printf("OK\n");
    }
    else{
      if(xmpc_node_num() == 0) printf("Error !\n");
      exit(1);
    }

  return 0;
}

File: tests/XMP/global-view/structure/F/gmove.f90
program gmove
  implicit none
  type b
     real(8):: a(10)
  end type b
!$xmp nodes p(2)
!$xmp template t(10)
!$xmp distribute t(block) onto p
!$xmp align b%a(i) with t(i)
!$xmp align a(i) with t(i)
  type(b):: c
  real(8) :: a(10)
  integer :: i
  logical :: flag = .false.

!$xmp loop on t(i)
  do i=1, 10
     c%a(i) = i;
     a(i) = i + 100;
  end do
    
!$xmp gmove
  a(1:3) = c%a(6:8)

  if (xmp_node_num() == 1) then
    if (a(1) == 6 .and. a(2) == 7 .and. a(3) == 8 .and. a(4) == 104 .and. a(5) == 105) then
       flag = .true.
    end if
 else
    flag = .true.    
 end if
    
 !$xmp reduction (.and.:flag)
 if (flag .eqv. .true.) then
    if (xmp_node_num() == 1) then
       write(*,*) "OK"
    end if
 else
    if (xmp_node_num() == 1) then
       write(*,*) "Error!"
    end if
    call exit(1)
 end if
end program gmove

